control-plane/p4RuntimeSerializer.cpp

```diff
@@ -181,6 +181,50 @@ struct Digest {
     const p4::P4DataTypeSpec* typeSpec;  // The format of the packed data.
 };
 
+struct Register {
+    const cstring name;  // The fully qualified external name of this register.
+    const IR::IAnnotated* annotations;  // If non-null, any annotations applied
```

Comment:
I think IAnnotated is the object which is annotated, not the annotations themselves. Perhaps this should be IR::Annotations instead.

Reply:
I'm just following what the existing code is doing. We actually store the declaration as `IR::IAnnotated` and then we have a `addAnnotations` method later (when creating the protobuf message) that retrieves the annotations.
I modified my gtest to use an annotation to make sure things were working correctly though :)